pkgs/development/compilers/llvm/common/llvm/default.nix

```diff
@@ -382,7 +382,7 @@ stdenv.mkDerivation (finalAttrs: {
     "-DCAN_TARGET_i386=false"
   ] ++ optionals ((stdenv.hostPlatform != stdenv.buildPlatform) && !(stdenv.buildPlatform.canExecute stdenv.hostPlatform)) [
```

Comment:
ditto, make it unconditional

Reply:
This doesn't look unconditional

Reply:
I would also move these ones up in the list now that they are unconditional

Reply:
Ha. My mistake here, and your bringing my attention to it made me spot the cross compiling logic immediately below. Is the whole PR bogus, again? It appears that with that kind of approach, we could allow LLVM to build its own tablegen within the derivation even for cross. (Perhaps with a bit of extension required to make it happen in the other non-`llvmPackages.llvm ` derivations).

I'm divided. Both ways have their pros and cons.

From a cursory read of the llvm cmake it looks to me as though passing CMAKE_CROSSCOMPILING may defeat LLVM_TABLEGEN. At least the worst this does is to cause a rebuild of tablegen within a cross-compiled LLVM, which seems to me something that can be fixed in a future PR.

Reply:
no I don't think the PR is bogus, we should strive for cross and native to work the same way, so it is good to make them both exclusively use the pre-built tools.

Reply:
> we could allow LLVM to build its own tablegen within the derivation even for cross.

It only works natively because it cannot compiler for two different platforms at once.

Reply:
> It only works natively because it cannot compile for two different platforms at once.

I'm not so sure? I think it actually does do that. CROSS_TOOLCHAIN_FLAGS_NATIVE is used to pass DCMAKE_C_COMPILER into a separate internal cmake invocation which is used to build the build tools.

One thing I don't understand is why nativeCC is defined as pkgsBuildBuild.targetPackages.stdenv.cc rather than just buildPackages.stdenv.cc; is there something interesting going on there?

Reply:
`pkgsBuildBuild.targetPackages.stdenv.cc` is the same as `buildPackages.stdenv.cc` since `pkgsBuildBuild.targetPackages` is `buildPackages` (which is the same as `pkgsBuildHost`).

Reply:
CMake invoking CMake like that (IIUC) is very janky and unlikely to work as well as a separate derivation because it bypasses setup hooks etc.

Reply:
Ack. So, sounds like we're settled on the current approach anyway. If there is something unaddressed please can you signpost it?

Reply:
I'm confused. I want this one to be unconditional too. Is something holding that up>

Reply:
Oh maybe the diff is wrong